notebook/services/contents/manager.py

```diff
@@ -440,10 +440,15 @@ def check_and_sign(self, nb, path=''):
         path : string
             The notebook's path (for logging)
         """
+        # Warning: The check_cells method removes the 
+        # "trusted"" key from the cells' metadata 
         if self.notary.check_cells(nb):
-            self.notary.sign(nb)
+            # We're checking the signature *after* the
+            # trusted keys have been removed
+            if not self.notary.check_signature(nb):
```

Comment:
I think this entire method should be left exactly as it was. Adding this check here is not correct because it prevents updating the signatures of trusted notebooks.